opengovsg/plumber
# PLU-318:  [TEMPLATES-11]: add template variable placeholders

## Problem
Users are unaware of what they can do with Plumber.

## Solution
Introduce templates to speed up their pipe creation and provide new ideas for them to automate. 

## Details
- Simulate template placeholder variable by following the step variable format to maintain backward compatibility: `step.fixed_uuid.template_label`
- For empty variable pill, add a different design to it, to confirm copy changes with Stacey

## Tests
For empty non-template variable:
1. Create webhook and test a variable: `test`
2. Use the variable in step 2.
3. Go back and test a different variable: `test1`
4. Click open step 2 and the pill should be updated to the empty variable (there should be an additional tooltip)

<img width="1085" alt="image" src="https://github.com/user-attachments/assets/1538b401-d4c3-4f43-ba5a-e4779f9223f2">


For template variable:
1. Test with any of the first 7 templates
2. Check the pipe and it should show the empty variable pill without any tooltip
<img width="921" alt="image" src="https://github.com/user-attachments/assets/8ea45d78-312c-4a7c-b5d0-3a21ef26504a">


For working variable:
Should still work accordingly
